test/eqDeep.js

```diff
@@ -0,0 +1,110 @@
+/* jshint typed: true */
+
+var assert = require('assert');
+
+var R = require('..');
+
+describe('eqDeep', function() {
+    var a = [];
+    var b = a;
+    it('tests for deep equality of its operands', function() {
+        assert.strictEqual(R.eqDeep(100, 100), true);
+        assert.strictEqual(R.eqDeep(100, '100'), false);
+        assert.strictEqual(R.eqDeep([], []), true);
+        assert.strictEqual(R.eqDeep(a, b), true);
+    });
+
+    it('handles objects', function() {
+        assert.strictEqual(R.eqDeep({}, {}), true);
+        assert.strictEqual(R.eqDeep({a:1, b:2}, {a:1, b:2}), true);
+        assert.strictEqual(R.eqDeep({a:2, b:3}, {b:3, a:2}), true);
+        assert.strictEqual(R.eqDeep({a:2, b:3}, {a:3, b:3}), false);
+        assert.strictEqual(R.eqDeep({a:2, b:3, c:1}, {a:2, b:3}), false);
+    });
+
+    var supportsSticky = false;
+    try { RegExp('', 'y'); supportsSticky = true; } catch (e) {}
+
+    var supportsUnicode = false;
+    try { RegExp('', 'u'); supportsUnicode = true; } catch (e) {}
+
+    it('handles regex', function() {
+        assert.strictEqual(R.eqDeep(/\s/, /\s/), true);
+        assert.strictEqual(R.eqDeep(/\s/, /\d/), false);
+        assert.strictEqual(R.eqDeep(/a/gi, /a/ig), true);
+        assert.strictEqual(R.eqDeep(/a/mgi, /a/img), true);
+        assert.strictEqual(R.eqDeep(/a/gi, /a/i), false);
+
+        /* jshint ignore: start */
+        // jscs:disable
+        // the sticky and unicode flags are not handled by jshint or jscs
+// this version of jscs doesn't support the jscs:disable/enable flag
+//        if (supportsSticky) {
+//            assert.strictEqual(R.eqDeep(/\s/y, /\s/y), true);
+//            assert.strictEqual(R.eqDeep(/a/mygi, /a/imgy), true);
+//        }
+//
+//        if (supportsUnicode) {
+//            assert.strictEqual(R.eqDeep(/\s/u, /\s/u), true);
+//            assert.strictEqual(R.eqDeep(/a/mugi, /a/imgu), true);
+//        }
+        // jscs:enable
+        /* jshint ignore: end */
```

Comment:
jshint and jscs don't recognize the 'sticky' and 'unicode' flags yet, and the version of jscs that we are using doesn't support the disable flag